packages/backend/src/open_payments/payment/incoming/routes.ts

```diff
@@ -132,22 +192,64 @@ function incomingPaymentToBody(
   received: bigint
 ) {
   const location = `${deps.config.publicHost}/incoming-payments/${incomingPayment.id}`
-  return {
+  const body = {
     id: location,
-    account: `${deps.config.publicHost}/pay/${incomingPayment.accountId}`,
-    amount: incomingPayment.amount.toString(),
-    assetCode: incomingPayment.account.asset.code,
-    assetScale: incomingPayment.account.asset.scale,
-    description: incomingPayment.description,
-    expiresAt: incomingPayment.expiresAt.toISOString(),
-    received: received.toString()
+    accountId: `${deps.config.publicHost}/pay/${incomingPayment.accountId}`,
+    state: incomingPayment.state.toLowerCase(),
+    receivedAmount: {
+      amount: received.toString(),
+      assetCode: incomingPayment.account.asset.code,
+      assetScale: incomingPayment.account.asset.scale
+    },
+    expiresAt: incomingPayment.expiresAt.toISOString()
   }
+
+  if (incomingPayment.incomingAmount) {
+    body['incomingAmount'] = {
+      amount: incomingPayment.incomingAmount.amount.toString(),
+      assetCode: incomingPayment.incomingAmount.assetCode,
+      assetScale: incomingPayment.incomingAmount.assetScale
+    }
+  }
+  if (incomingPayment.description)
+    body['description'] = incomingPayment.description
+  if (incomingPayment.externalRef)
+    body['externalRef'] = incomingPayment.externalRef
+  // workaround: will be removed with update to ilp-pay:0.4.0-alpha.2
+  body['receiptsEnabled'] = false
+  return body
 }
 
-function tryParseAmount(amount: unknown): bigint | null {
-  try {
-    return BigInt(amount)
-  } catch (_) {
-    return null
+function parseAmount(amount: unknown): Amount | undefined {
+  if (amount === undefined) return amount
+  if (
+    typeof amount !== 'object' ||
+    amount === null ||
+    typeof amount['amount'] !== 'number' ||
+    amount['amount'] <= 0 ||
```

Comment:
`amount['amount']` can also (should?) be a string.
Maybe just lean on the `amount: BigInt(amount['amount'])` below, and check that the amount is positive in `IncomingPaymentService.create`?

Reply:
It’s an integer in the [spec](https://docs.openpayments.guide/reference/create-incoming-payment). I can move the check to the service, though.

Reply:
I see the examples have it as a number, but 64-bit integers are represented as strings in JSON.

Reply:
Oh, I missed the part of it being 64-bit integers (one should always read the text). I'll update the spec as well.

Reply:
But isn't that a JS/TS problem, and not a JSON problem? JSON doesn't care whether it's a number or a string, but when we try to parse it, JS/TS has issues. 
I changed it to be a string but I think we should put in the spec that amount is a stringified 64-bit integer. Or we add 2 values like [Twitter does with "id" and "id_str"](https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/object-model/example-payloads).